charts/kong/templates/admission-webhook.yaml

```diff
@@ -227,6 +227,16 @@ spec:
     port: 443
     protocol: TCP
     targetPort: webhook
+  {{- if and .Values.ingressController.enabled (semverCompare ">= 2.0.0" (include "kong.effectiveVersion" .Values.ingressController.image)) }}
+  - name: cmetrics
+    port: 10255
+    protocol: TCP
+    targetPort: cmetrics
+  - name: status
+    port: 10254
+    protocol: TCP
+    targetPort: cstatus
+  {{- end }}
```

Comment:
Instead of adding ports to webhook service, I'd suggest that we add a separate service which will be only responsible for metrics. This way it will be easier for users to create RBAC rules around and it will not require to run webhook (some users don't) for those that only want metrics.

WDYT?

Reply:
Thanks for the response. 

> This way it will be easier for users to create RBAC rules around and it will not require to run webhook (some users don't) for those that only want metrics.

For GW services these ports are already added to the service object correctly (and work as expected). However these ports are missing _just_ on the controller (webhook) service (as the service object is created in a distinct manner when compared to the GW services using the service template). 
Therefore not opposed to creating a distinct service, but it will still be tightly coupled to the webhook service. 

With the above context, is the recommendation still to add a distinct service for this purpose?

Reply:
Have added in the additional service (with the same restrictions as the webhook service has)